src/v/serde/time_point.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "serde.h"
+
+#include <seastar/core/lowres_clock.hh>
+
+namespace serde {
+
+// Only system clocks time points are serializable. This is a loose check
+// but good enough, we assume any impls with to_time_t and from_time_t
+// are system clocks.
+template<class T>
+concept is_system_clock = requires {
+    T::clock::to_time_t;
```

Comment:
It's unfortunate that libc++ doesn't seem to have [std::is_clock`](https://en.cppreference.com/w/cpp/chrono/is_clock), but those checks (or a standalone concept) could be added, too, for completeness.